docs/sdks/typescript/index.md

```diff
@@ -55,927 +67,740 @@ Each SpacetimeDB client depends on some bindings specific to your module. Create
-| [`SpacetimeDBClient.constructor`](#spacetimedbclient-constructor) | Creates a new `SpacetimeDBClient` database client.                       |
-| Properties                                                        |
-| [`SpacetimeDBClient.identity`](#spacetimedbclient-identity)       | The user's public identity.                                              |
-| [`SpacetimeDBClient.live`](#spacetimedbclient-live)               | Whether the client is connected.                                         |
-| [`SpacetimeDBClient.token`](#spacetimedbclient-token)             | The user's private authentication token.                                 |
-| Methods                                                           |                                                                          |
-| [`SpacetimeDBClient.connect`](#spacetimedbclient-connect)         | Connect to a SpacetimeDB module.                                         |
-| [`SpacetimeDBClient.disconnect`](#spacetimedbclient-disconnect)   | Close the current connection.                                            |
-| [`SpacetimeDBClient.subscribe`](#spacetimedbclient-subscribe)     | Subscribe to a set of queries.                                           |
-| Events                                                            |                                                                          |
-| [`SpacetimeDBClient.onConnect`](#spacetimedbclient-onconnect)     | Register a callback to be invoked upon authentication with the database. |
-| [`SpacetimeDBClient.onError`](#spacetimedbclient-onerror)         | Register a callback to be invoked upon a error.                          |
+A connection to a remote database is represented by the `module_bindings.DbConnection` type. This type is generated per-module, and contains information about the types, tables and reducers defined by your module.
 
-## Constructors
+| Name                                                      | Description                                                                                      |
+|-----------------------------------------------------------|--------------------------------------------------------------------------------------------------|
+| [Connect to a module](#connect-to-a-module)               | Construct a `DbConnection`.                                                                      |
+| [Access tables and reducers](#access-tables-and-reducers) | Access subscribed rows in the client cache, request reducer invocations, and register callbacks. |
 
-### `SpacetimeDBClient` constructor
 
-Creates a new `SpacetimeDBClient` database client and set the initial parameters.
+### Connect to a module
 
-```ts
-new SpacetimeDBClient(host: string, name_or_address: string, auth_token?: string, protocol?: "binary" | "json")
+```typescript
+DbConnection.builder(): DbConnectionBuilder
```

Comment:
```suggestion
builder(): DbConnectionBuilder
```

Or

```ts
class DbConnectionBuilder {
  public withUri(uri: string): DbConnectionBuilder;
}

If you prefer. The above implies that it is a class method.

Reply:
Let's do the latter I guess.

```suggestion
class DbConnection {
  public builder(): DbConnectionBuilder;
}
```

Reply:
I see what you mean. This one *is* what I would call a static method, or an associated function, which takes the class as a receiver. But I used this same syntax for what I would call methods, which take an instance as a receiver. What syntax would you use for each of these things, in general?

Reply:
Oh, whoops, you're right. It's fine for this particular one, but confusing for the others. For static what you have here is fine:

```ts
DbConnection.builder(): DbConnectionBuilder
```

For methods let's do this:

```ts
class DbConnectionBuilder {
  public withUri(uri: string): DbConnectionBuilder
}
```

Reply:
If we're going to use the `class` syntax, let's write static methods as static methods, [since JS apparently supports that](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/static).